src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -45,10 +45,13 @@ end
 const MaybeTupledData = Union{OffsetArray, NTuple{<:Any, OffsetArray}}
 
 "Fill halo regions in ``x``, ``y``, and ``z`` for a given field's data."
-function fill_halo_regions!(c::MaybeTupledData, boundary_conditions, indices, loc, grid, args...; kwargs...)
-
+function fill_halo_regions!(c::MaybeTupledData, boundary_conditions, indices, loc, grid, args...; fill_boundary_normal_velocities = true, kwargs...)
     arch = architecture(grid)
 
+    if fill_boundary_normal_velocities
+        fill_open_boundary_regions!(c, boundary_conditions, loc, grid, args...; kwargs...)
+    end
```

Comment:
Are open boundary conditions filled before everything else here?

Reply:
And this was reflected in regression tests failures when I had the open halo fills after the periodic before